# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/loader/ScriptPreloader.cpp
# Commit: 8f3257e6d425
# Full Hash: 8f3257e6d4255e617412549cf613353b3477215f
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2021-10-14 03:02:57
# Description:
#   Bug 1734612 - Use soft error and stop reading startup cache file if the content is unexpected. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D128320
# ==============================================================================

diff -r f236b1ba46f0 -r 8f3257e6d425 js/xpconnect/loader/ScriptPreloader.cpp
--- a/js/xpconnect/loader/ScriptPreloader.cpp	Wed Oct 13 16:12:51 2021 +0000
+++ b/js/xpconnect/loader/ScriptPreloader.cpp	Wed Oct 13 16:13:00 2021 +0000
@@ -537,7 +537,9 @@
       MOZ_RELEASE_ASSERT(script);
 
       auto scriptData = data + script->mOffset;
-      MOZ_RELEASE_ASSERT(JS::IsTranscodingBytecodeAligned(scriptData.get()));
+      if (!JS::IsTranscodingBytecodeAligned(scriptData.get())) {
+        return Err(NS_ERROR_UNEXPECTED);
+      }
 
       if (scriptData + script->mSize > end) {
         return Err(NS_ERROR_UNEXPECTED);
